cache/cache.cpp

```diff
@@ -239,22 +239,22 @@ static int n2_fill_lookup_response_without_fd(struct dnet_net_state *st,
 		dnet_checksum_data(st->n, data, size, checksum.data(), checksum.size());
 	}
 
-	response.reset(new(std::nothrow) n2::lookup_response(*cmd,
-	                                                     0, // record_flags
-	                                                     0, // user_flags
-	                                                     {}, // path
-	                                                     {0, 0}, // json_timestamp
-	                                                     0, // json_offset
-	                                                     0, // json_size
-	                                                     0, // json_capacity
-	                                                     {}, // json_checksum
-	                                                     timestamp, // data_timestamp
-	                                                     0, // data_offset
-	                                                     size, // data_size
-	                                                     std::move(checksum) // data_checksum
+	response.reset(new n2::lookup_response(*cmd,
+	                                       0, // record_flags
+	                                       0, // user_flags
+	                                       {}, // path
+	                                       {0, 0}, // json_timestamp
+	                                       0, // json_offset
+	                                       0, // json_size
+	                                       0, // json_capacity
+	                                       {}, // json_checksum
+	                                       timestamp, // data_timestamp
+	                                       0, // data_offset
+	                                       size, // data_size
+	                                       std::move(checksum) // data_checksum
 	));
 
-	return response ? 0 : -ENOMEM;
+	return 0;
```

Comment:
How about make response a return value?
```cpp
return std::unique_ptr<n2::lookup_response>(new n2::lookup_response(...
```

Reply:
Direct-return will be very doddering signature with our no-throw politic :-( One has only add any C-style stuff in implementation, and we shall have to revert it. I current case I found a bug in `n2_fill_lookup_response_without_fd`: there called `dnet_checksum_data` without checking result. So if we check result, we cannot do as you suggest.

If we don't plan to check result of `dnet_checksum_data`, I rework the signature.

Reply:
`dnet_send_file_info_ts_without_fd` is just reworked `n2_fill_lookup_response_without_fd`.
`dnet_checksum_data` tries to calculate checksum and we skip error there.
I'm not sure, may be it's unexpected behaviour, but I saved it.

Reply:
I found, method `dnet_local_digest_transform` always returns 0.

Reply:
(for the time being left it at that)

Reply:
upd. Fixed

Reply:
upd. Fixed